### PR TITLE
Oana ma

### PR DESCRIPTION
**Pull request by laburnumT**

### CODE REVIEW

**Comment by laburnumT** [comment]
Could you squash the commits into one and update the commit message?

File: src/Eval.rsc [hunk 1]
```diff
@@ -24,12 +22,92 @@ alias VEnv = map[str name, Value \value];
 data Input
   = input(str question, Value \value);
 
+Value getDefaultValue(AType typeName) {
+  switch (typeName) {
```
Comment (laburnumT) [inline src/Eval.rsc:26]: Curly braces around cases

File: src/Test.rsc [hunk 2]
```diff
@@ -17,3 +18,39 @@ void testMain(loc input) {
...
+    case bool b:
+      return vbool(b);
+    case str s:
+      return vstr(s);
+    default:
+      throw "Unsupported value <val>";
+  }
+}
+
+void testEval() {
```
Comment (laburnumT) [inline src/Test.rsc]: I would move this functionality into testMain.

Now it's just duplicating a bunch of work.

File: src/Test.rsc [hunk 3]
```diff
@@ -17,3 +18,39 @@ void testMain(loc input) {
...
+  // modify these ones for testing
+  loc location = |cwd:///examples/test1.myql|;
+  map[str, value] venv_map = ("hasSoldHouse": true, "sellingPrice": 300, "privateDebt": 20);
```
Comment (laburnumT) [inline src/Test.rsc]: Shouldn't this be created by the initialEnv function?

File: src/Test.rsc [hunk 4]
```diff
@@ -17,3 +18,39 @@ void testMain(loc input) {
...
+  set[Message] msgs = check(ast, tenv, g.useDef);
+
+  VEnv env = initialEnv(ast);
+
+  for(venv_name <- venv_map) {
+    Value val = getValue(venv_map[venv_name]);
+    env = eval(ast, input(venv_name, val), env);
+  }
+
+  println(env);
```
Comment (laburnumT) [inline src/Test.rsc]: Remove the final println.

**Comment by laburnumT** [comment]
Also please do a rebase on master.

File: src/Eval.rsc (hunk 1 above)
Comment (laburnumT) [reply src/Eval.rsc:26]: A regex that should work for this is:

```
s/(\s*)case (.*):\n(.*)/\1case \2: {\n\3\1}/
```

File: src/Test.rsc (hunk 3 above)
Comment (oana-ma) [reply src/Test.rsc]: @laburnumT This was before compiling the forms was available. The venv_map is filled with variables and their values for test purposes, e.g. to see how the forms respond to certain user input.

Comment (laburnumT) [reply src/Test.rsc]: Ok. Ignore my comment then.

**Comment by laburnumT** [comment]
Signed-off-by: laburnumT <laburnumtec@gmail.com>

**Comment by laburnumT** [comment]
If you rebase and squash it LGTM.

I can do it for you too.

**Pull request merged**